### PR TITLE
html-proofer: 3.18.8 -> 5.0.8 

### PR DESCRIPTION
**Pull request by alejandro-angulo**
## Description of changes

There were some big changes made from version 3 -> 4 (and this change will bump us all the way to major version 5): https://github.com/gjtorikian/html-proofer/blob/main/UPGRADING.md

I had to run `bundle config set force_ruby_platform true` for this to build directly. I think this would be equivalent to running `env BUNDLE_FORCE_RUBY_PLATFORM=true bundle lock`. Not sure if the updater script should be updated [here](https://github.com/NixOS/nixpkgs/blob/e62118a3f914e1764bbdfdaed132734c0cbfee9d/pkgs/development/ruby-modules/bundler-update-script/default.nix#L21).

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by AndersonTorres** [comment]
"Bump" what?

**Review by mohe2015 — COMMENTED** [review]
Please adjust the commit message according to the contribution guidelines.

**Comment by alejandro-angulo** [comment]
Ah sorry i had amended the commit message locally but forgot to push it :sweat_smile:

File: nixos/doc/manual/release-notes/rl-2305.section.md [hunk 1]
```diff
@@ -363,6 +363,8 @@ In addition to numerous new and updated packages, this release has the following
 
 - The `nerdfonts` package has been updated to major version 3, which includes potential [breaking changes](https://github.com/ryanoasis/nerd-fonts/releases/tag/v3.0.0).
 
+- The `html-proofer` package has been updated from major version 3 to major version 5, which includes [breaking changes](https://github.com/gjtorikian/html-proofer/blob/v5.0.8/UPGRADING.md).
```
Comment (AndersonTorres) [inline nixos/doc/manual/release-notes/rl-2305.section.md:366]: 1. write this in rl-2311
2. Split in a separate commit, with title  `rl-2311: mention upgrade of html-proofer to major version 5`

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: nixos/doc/manual/release-notes/rl-2305.section.md (hunk 1 above)
Comment (alejandro-angulo) [reply nixos/doc/manual/release-notes/rl-2305.section.md]: Ah I'm making a bunch of noob mistake. I appreciate the quick review though :+1: 

lmk if anything else should be changed

**Comment by AndersonTorres** [comment]
Result of `nixpkgs-review pr 248984` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>html-proofer</li>
  </ul>
</details>
